src/tools/tests/kotlin-codegen-test-suite.ts

```diff
@@ -0,0 +1,110 @@
+/**
+ * @license
+ * Copyright (c) 2019 Google Inc. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * Code distributed by Google as part of this project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import {SchemaGraph} from '../schema2graph.js';
+import {generateSchema} from '../kotlin-schema-generator.js';
+import {ManifestCodegenUnitTest} from '../codegen-unit-test.js';
+import {KTExtracter} from '../kotlin-refinement-generator.js';
+import {CodegenUnitTest} from '../codegen-unit-test.js';
+import {Manifest} from '../../runtime/manifest.js';
+import {Flags} from '../../runtime/flags.js';
+import {KotlinEntityGenerator} from '../kotlin-entity-generator.js';
+import {Schema2Kotlin} from '../schema2kotlin.js';
+import {generateConnectionSpecType, generateType} from '../kotlin-type-generator.js';
+
+/**
+ * A Suite of unit tests for Kotlin Codegen.
+ * The test data is inside .cgtest files.
+ */
+export const testSuite: CodegenUnitTest[] = [
+  new class extends ManifestCodegenUnitTest {
+    constructor() {
+      super(
+        'Kotlin Schema Generation',
+        'kotlin-schema-generation.cgtest'
+      );
+    }
+    async computeFromManifest({particles}) {
+      const graph = new SchemaGraph(particles[0]);
+      return Promise.all(graph.nodes.map(async node => {
+        await Promise.all([node, ...node.refs.values()].map(n => n.calculateHash()));
+        return generateSchema(node.schema);
+      }));
+    }
+  }(),
+  new class extends CodegenUnitTest {
+    constructor() {
+      super(
+        'Kotlin Refinement Generator',
+        'kotlin-refinement-generator.cgtest'
+      );
+    }
+    async compute(input: string) {
```

Comment:
Why can't this call `computeFromManifest`? Is it so you can use flags in parsing?

Reply:
`computeFromManifest` could take a `flagsSettings` argument for this case... this is an optional nit, really.

Reply:
Yup, exactly... flags are needed at parse time :/ 

Looking into it now.

Reply:
Done, allowed overwriting flags in the constructor.